integrations/go/workload-executor.go

```diff
@@ -0,0 +1,306 @@
+
+func executeInsertOne(coll *mongo.Collection, args bson.Raw) (*mongo.InsertOneResult, error) {
+	doc := emptyDoc
+	opts := options.InsertOne()
+
+	elems, _ := args.Elements()
+	for _, elem := range elems {
+		key := elem.Key()
+		val := elem.Value()
+
+		switch key {
+		case "document":
+			doc = val.Document()
+		default:
+			str := fmt.Sprintf("unrecognized insertOne option: %v", key)
+			panic(str)
+		}
+	}
+
+	return coll.InsertOne(context.Background(), doc, opts)
+}
+
+func executeFind(coll *mongo.Collection, args bson.Raw) (*mongo.Cursor, error) {
+	filter := emptyDoc
+	opts := options.Find()
+
+	elems, _ := args.Elements()
+	for _, elem := range elems {
+		key := elem.Key()
+		val := elem.Value()
+
+		switch key {
+		case "filter":
+			filter = val.Document()
+		case "sort":
+			opts = opts.SetSort(val.Document())
+		default:
+			str := fmt.Sprintf("unrecognized find option: %v", key)
+			panic(str)
+		}
+	}
+
+	return coll.Find(context.Background(), filter, opts)
+}
+
+// create an update document or pipeline from a bson.RawValue
+func createUpdate(updateVal bson.RawValue) (interface{}, error) {
+	switch updateVal.Type {
+	case bson.TypeEmbeddedDocument:
+		return updateVal.Document(), nil
+	case bson.TypeArray:
+		var updateDocs []bson.Raw
+		docs, _ := updateVal.Array().Values()
+		for _, doc := range docs {
+			updateDocs = append(updateDocs, doc.Document())
+		}
+
+		return updateDocs, nil
+	default:
+		str := fmt.Sprintf("unrecognized update type: %v", updateVal.Type)
+		panic(str)
+	}
+
+	return nil, nil
+}
+
+func executeUpdateOne(coll *mongo.Collection, args bson.Raw) (*mongo.UpdateResult, error) {
+	filter := emptyDoc
+	var update interface{} = emptyDoc
+	var err error
+	opts := options.Update()
+
+	elems, _ := args.Elements()
+	for _, elem := range elems {
+		key := elem.Key()
+		val := elem.Value()
+
+		switch key {
+		case "filter":
+			filter = val.Document()
+		case "update":
+			update, err = createUpdate(val)
+			if err != nil {
+				return nil, err
+			}
+		default:
+			str := fmt.Sprintf("unrecognized updateOne option: %v", key)
+			panic(str)
+		}
+	}
+	if opts.Upsert == nil {
+		opts = opts.SetUpsert(false)
+	}
+
+	return coll.UpdateOne(context.Background(), filter, update, opts)
+}
+
+func verifyInsertOneResult(actualResult *mongo.InsertOneResult, expectedResult interface{}) bool {
+	if expectedResult == nil {
+		return true
+	}
+
+	var expected mongo.InsertOneResult
+	if bson.Unmarshal(expectedResult.(bson.Raw), &expected) != nil {
+		return false
+	}
+
+	expectedID := expected.InsertedID
+	if f, ok := expectedID.(float64); ok && f == math.Floor(f) {
+		expectedID = int32(f)
+	}
+
+	return expectedID == nil || (actualResult != nil && expectedID == actualResult.InsertedID)
+}
+
+func verifyCursorResult(cur *mongo.Cursor, result interface{}) bool {
+	if result == nil {
+		return true
+	}
+
+	if cur == nil {
+		return false
+	}
+
+	defer func() {
```

Comment:
We should be checking the error from `cur.Close` in the defer. We can probably just panic if it errors.

Reply:
done